charts/newrelic-pixie/templates/deployment.yaml

```diff
@@ -0,0 +1,69 @@
+{{- if (include "newrelic-pixie.areValuesValid" .) }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "newrelic-pixie.fullname" . }}
+  namespace: {{ template "newrelic-pixie.namespace" . }}
+  labels:
+    {{- include "newrelic-pixie.labels" . | nindent 4 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ template "newrelic-pixie.name" . }}
+      release: {{.Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ template "newrelic-pixie.name" . }}
+        release: {{.Release.Name }}
+    spec:
+      containers:
+      - name: {{ template "newrelic-pixie.name" . }}
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+        imagePullPolicy: "{{ .Values.image.pullPolicy }}"
+        env:
+        - name: CLUSTER_NAME
+          value: {{ template "newrelic-pixie.cluster" . }}
+        - name: NR_LICENSE_KEY
+          valueFrom:
+            secretKeyRef:
+              {{- if (include "newrelic-pixie.licenseKey" .) }}
+              name: {{ template "newrelic-pixie.fullname" . }}-secrets
+              key: newrelicLicenseKey
+              {{- else }}
+              name: {{ include "newrelic-pixie.customSecretName" . }}
+              key: {{ include "newrelic-pixie.customSecretLicenseKey" . }}
+              {{- end }}
+        - name: PIXIE_API_KEY
+          valueFrom:
+            secretKeyRef:
+              {{- if (include "newrelic-pixie.apiKey" .) }}
+              name: {{ template "newrelic-pixie.fullname" . }}-secrets
+              key: pixieApiKey
+              {{- else }}
+              name: {{ include "newrelic-pixie.customSecretName" . }}
+              key: {{ include "newrelic-pixie.customSecretApiKeyKey" . }}
+          {{- end }}
+        - name: PIXIE_CLUSTER_ID
+          {{- if .Values.pixieClusterId }}
```

Comment:
Where does this value come from?

Reply:
It's intentionally not documented. We should load this from a secret that already exists in the cluster. But in the CI it doesn't exist, so I need another way to send it there.

Reply:
```suggestion
          {{- if .Values.pixieClusterId }}
          {{- /* .Values.pixieClusterId is only used for CI tests, for regular operation it's mounted from pixie secrets. */ }}
```

Reply:
Hi, i'm deploying this new integration, and with no value for pixieClusterId, where is this value supposed to come from? because the pod newrelic-bundle-newrelic-pixie-* is not being able to run due to the following errors:

- At first pl-cluster-secrets is not present
- then, pl-cluster-secrets does not contain cluster-id (Error: couldn't find key cluster-id in Secret newrelic/pl-cluster-secrets)

Reply:
Hey @ulm0! This key should appear in the secret after the Pixie cloud connector pod correctly authenticates and exchange some data with the Pixie Cloud.